lib/cli/github_settings/do_set_branch_protection.js

```diff
@@ -152,33 +163,86 @@ function do_set_branch_protection(subcmd, opts, args, cb) {
                         return;
                     }
 
-                    var reqOpts = {
-                        owner: 'joyent',
-                        repo: repoName,
-                        branch: 'master',
+                    var reqOpts = {};
+                    var lockMessage = '';
 
-                        enforce_admins: true,
-                        required_pull_request_reviews: {
-                            dismissal_restrictions: {},
-                            dismiss_stale_reviews: true,
-                            require_code_owner_reviews: false,
-                            required_approving_review_count: 1
-                        },
-                        restrictions: null,
-                        // Don't muck with this value yet, until MANTA-4598 is
-                        // decided and complete.
-                        required_status_checks: ctx.masterProt
-                            ? ctx.masterProt.required_status_checks
-                            : null
-                    };
+                    if (ctx.unlockUsers) {
+                        lockMessage =
+                            'Allowing only ' +
+                            ctx.unlockUsers.join(', ') +
+                            ' to push to "master" ' +
+                            'branch of %s ' +
+                            '(https://github.com/joyent/%s/settings/branches)';
+                        reqOpts = {
+                            owner: 'joyent',
+                            repo: repoName,
+                            branch: 'master',
+
+                            enforce_admins: true,
+                            required_pull_request_reviews: null,
+                            restrictions: {
+                                users: ctx.unlockUsers,
+                                teams: []
+                            },
+                            required_status_checks: null
+                        };
+                        // fml
+                        if (
+                            ctx.unlockUsers &&
+                            ctx.masterProt &&
+                            ctx.masterProt.required_status_checks &&
+                            ctx.masterProt.required_status_checks
+                                .contexts &&
+                            ctx.masterProt.required_status_checks.contexts
+                                .length > 0
+                        ) {
+                            lockMessage = lockMessage + '\n' +
+                                'The following status checks were ' +
+                                'disabled:\n ' +
+                                ctx.masterProt.required_status_checks.contexts.join(' \n');
```

Comment:
4-space indent in the lockMessage please

Reply:
By that, I think you mean break the dotted-path up to a further 4-space indent? I'll do that.

Reply:
No, I meant to have a 4-space indent in the printed message:

```
    'disabled:\n    ' +
    ctx.masterProt.required_status_checks.contexts.join('\n    ');
```